src/main/java/io/appium/java_client/AppiumDriver.java

```diff
@@ -129,6 +127,35 @@ public AppiumDriver(HttpClient.Factory httpClientFactory, Capabilities desiredCa
 
     public AppiumDriver(Capabilities desiredCapabilities) {
         this(AppiumDriverLocalService.buildDefaultService(), desiredCapabilities);
+
+    }
+
+    /**
+     * For creating a client connected to an existing Appium session.
+     * The other constructors all create a new session, sending the POST /session request.
+     * This constructor does not.
+     */
+    public AppiumDriver(URL remoteAddress, String sessionId) {
+        this.setSessionId(sessionId);
+        
+        AppiumCommandExecutor executor = new AppiumCommandExecutor(MobileCommand.commandRepository, remoteAddress);
+        executor.setCommandCodec(new AppiumW3CHttpCommandCodec());
+        executor.setResponseCodec(Dialect.W3C.getResponseCodec());
+        executor.configureW3CMode();
```

Comment:
both `configureW3CMode` and this method call `setCommandCodec`

Reply:
Good point. I refactored at the last second. Will remove that line.